libvmi/driver/xen/xen.c

```diff
@@ -2504,6 +2504,39 @@ xen_set_vcpureg_arm(
 }
 #endif
 
+status_t
+xen_request_page_fault(
+    vmi_instance_t vmi,
+    unsigned long vcpu,
+    uint64_t virtual_address,
+    uint32_t error_code)
+{
+#ifdef ENABLE_SAFETY_CHECKS
+#if defined(I386) || defined(X86_64)
+    uint64_t value;
+
+    if (xen_get_vcpureg_hvm(vmi, &value, SS_ARBYTES, vcpu) == VMI_FAILURE)
+        return VMI_FAILURE;
+
+    if (((value >> 5) & 3) != 3) {
```

Comment:
Could you please explain this ?

Reply:
Sorry this is post merge.

Reply:
@thatreguy nothing wrong with commenting after a merge in my view - sometimes issues are noticed only after the fact, and we might as well place those comments with the related change.

Anyway, I'm generally a fan of macros, but during review I saw this as fairly succinct and clear in context due to the proximity of the register and the error message.  We're certainly open to contributions if you had something in mind to improve documentation/readability.